feat(profiles): bound dependencies, remove uv.lock files, improve init output (#2754)
canonical/charmcraft: docs/reference/files/uv-lock-file.rst

    `uv | The lockfile
    <https://docs.astral.sh/uv/concepts/projects/layout/#the-lockfile>`_

Charmcraft doesn't create this file. This file is required if your charm uses the

> :ref:`craft_parts_uv_plugin`. It's also required by the tox-uv plugin. For more

Reviewer: Optional:
```suggestion
Charmcraft doesn't create this file. This file is required by the tox-uv plugin and
charms with environments managed by the :ref:`craft_parts_uv_plugin`. For more
```

information, see :ref:`tox-ini-file`.

Use ``uv lock`` to create this file and update it if you change your charm's
dependencies. Alternatively, you can update this file by using ``uv add`` and

> ``uv remove`` to manage dependencies in the :ref:`pyproject-toml-file`.

Reviewer: Optional:
```suggestion
To create or update this file, run the ``uv lock`` command. Alternatively,
you can update this file by running ``uv add`` or ``uv remove`` to
manage dependencies in the :ref:`pyproject-toml-file`.
```

You shouldn't manually edit this file.

> For 12-factor app profiles targeting Ubuntu 24.04 LTS or lower: This file isn't

Reviewer: Optional:
```suggestion
For 12-factor app charms targeting Ubuntu 24.04 LTS or lower: This file isn't
```

    <https://docs.astral.sh/uv/concepts/projects/layout/#the-lockfile>`_

This file is required if your charm uses the :ref:`craft_parts_uv_plugin`. It's also
required by the tox-uv plugin. For more information, see :ref:`tox-ini-file`.

> Charmcraft doesn't create this file.

Reviewer 2: Does this need to be a paragraph of its own? It seems like it could be the intro for the next one.
